src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -0,0 +1,959 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// A source method that can have attributes, including a member method, accessor, or local function.
+    /// </summary>
+    internal abstract class SourceMethodSymbolWithAttributes : SourceMethodSymbol, IAttributeTargetSymbol
+    {
+        private CustomAttributesBag<CSharpAttributeData> _lazyCustomAttributesBag;
+        private CustomAttributesBag<CSharpAttributeData> _lazyReturnTypeCustomAttributesBag;
+
+        // some symbols may not have a syntax (e.g. lambdas, synthesized event accessors)
+        protected readonly SyntaxReference syntaxReferenceOpt;
+        protected SourceMethodSymbolWithAttributes(SyntaxReference syntaxReferenceOpt)
+        {
+            this.syntaxReferenceOpt = syntaxReferenceOpt;
+        }
+
+        internal CSharpSyntaxNode GetBinderContextNode()
+        {
+            CSharpSyntaxNode syntaxNode = this.SyntaxNode;
+            CSharpSyntaxNode contextNode;
+
+            switch (syntaxNode)
+            {
+                case ConstructorDeclarationSyntax constructor:
+                    contextNode = constructor.Initializer ?? (CSharpSyntaxNode)constructor.Body ?? constructor.ExpressionBody;
```

Comment:
>contextNode [](start = 20, length = 11)

Minor: Perhaps simply `return expr;` from each branch rather than assigning to a local.